x-pack/plugins/observability_solution/entity_manager/server/lib/entities/update_entity_definition.ts

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import semver from 'semver';
+import { EntityDefinition } from '@kbn/entities-schema';
+import { findEntityDefinitions } from './find_entity_definition';
+import { uninstallEntityDefinition } from './uninstall_entity_definition';
+import { installAndStartDefinition } from './install_entity_definition';
+import { EntityManagerServerSetup } from '../../types';
+import { checkIfEntityDiscoveryAPIKeyIsValid, readEntityDiscoveryAPIKey } from '../auth';
+import { getClientsFromAPIKey } from '../utils';
+
+export async function updateBuiltInEntityDefinitions({
+  definitions,
+  server,
+}: {
+  definitions: EntityDefinition[];
+  server: EntityManagerServerSetup;
+}) {
+  const { logger } = server;
+  const apiKey = await readEntityDiscoveryAPIKey(server);
+  if (!apiKey) {
+    logger.debug('No API key found, skipping built-in definition upgrade');
+    return;
+  }
+
+  const isValid = await checkIfEntityDiscoveryAPIKeyIsValid(server, apiKey);
+  if (!isValid) {
+    logger.debug(`Stored API key is not valid, skipping built-in definition upgrade`);
```

Comment:
is this actually an error condition?

Reply:
Yes, I've updated the log level. Also added a return condition in the `GET /enablement` route that verifies if the installed definitions are not aligned with the latest version (ie failed upgrade)